src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1410,6 +1412,64 @@ internal async Task HandleUpdateAgentNotebookRequest(UpdateAgentNotebookParams p
             });
         }
 
+        internal async Task HandleUpdateAgentNotebookRunNameRequest(UpdateAgentNotebookRunNameParams parameters, RequestContext<ResultStatus> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                var result = new ResultStatus();
+                try
+                {
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                                                parameters.OwnerUri,
+                                                out connInfo);
+                    // Calling update helper function
+                    await AgentNotebookHelper.UpdateMaterializedNotebookName(
+                        connInfo,
+                        parameters.MaterializedId,
+                        parameters.TargetDatabase,
+                        parameters.MaterializedNotebookName);
+                    result.Success = true;
+                }
+                catch (Exception e)
+                {
+                    result.Success = false;
+                    result.ErrorMessage = e.ToString();
+
+                }
+                await requestContext.SendResult(result);
+            });
+        }
+
+        internal async Task HandleUpdateAgentNotebookRunPinRequest(UpdateAgentNotebookRunPinParams parameters, RequestContext<ResultStatus> requestContext)
```

Comment:
So we are considering pinned to be a attribute of the notebook and not of the user's preference?